### PR TITLE
Use `noble` bytes conversion utilities internally

### PR DESCRIPTION
**Pull request by acolytec3**
Addresses #3693 since the noble hex to bytes conversion utility is about 2x ours.

### CODE REVIEW

File: packages/util/src/bytes.ts [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import { getRandomBytesSync } from 'ethereum-cryptography/random.js'
 // eslint-disable-next-line no-restricted-imports
-import { bytesToHex as _bytesToUnprefixedHex } from 'ethereum-cryptography/utils.js'
+import {
+  bytesToHex as _bytesToUnprefixedHex,
```
Comment (paulmillr) [inline packages/util/src/bytes.ts:4]: it's not unprefixed - it allows 0x, but doesn't mandate it

**Comment by holgerd77** [comment]
Browser tests still failing

**Comment by holgerd77** [comment]
Update: just read up on Discord, let me know if I should still deactivate the browser tests.

File: packages/util/src/bytes.ts [hunk 2]
```diff
@@ -73,6 +54,9 @@ export const bytesToBigInt = (bytes: Uint8Array, littleEndian = false): bigint =
   if (littleEndian) {
     bytes.reverse()
   }
+
+  // Ensure that bytes are actually bytes (and not just an array of numbers)
+  // bytes = Uint8Array.from(bytes)
```
Comment (jochem-brouwer) [inline packages/util/src/bytes.ts:59]: This is commented out?

File: packages/client/test/net/protocol/ethprotocol.spec.ts [hunk 3]
```diff
@@ -70,7 +70,7 @@ describe('[EthProtocol]', () => {
       'encode status',
     )
     const status = p.decodeStatus({
-      chainId: [0x01],
+      chainId: Uint8Array.from([0x01]),
```
Comment (jochem-brouwer) [inline packages/client/test/net/protocol/ethprotocol.spec.ts:73]: Why is this change now necessary?

Comment (paulmillr) [reply packages/client/test/net/protocol/ethprotocol.spec.ts:73]: the function accepted arrays before. Now it is limited to u8a

File: packages/util/src/bytes.ts (hunk 2 above)
Comment (acolytec3) [reply packages/util/src/bytes.ts:59]: Meant to remove it entirely.  It breaks a bunch of VM tests and doesn't solve anything other than a couple of client tests that were already incorrect.

File: packages/client/test/net/protocol/ethprotocol.spec.ts (hunk 3 above)
Comment (acolytec3) [reply packages/client/test/net/protocol/ethprotocol.spec.ts:73]: Furthermore, this property is supposed to be a Uint8Array if you look at the types.

**Review by scorbajio — APPROVED** [review]
LGTM

File: config/vitest.config.browser.mts [hunk 4]
```diff
@@ -22,6 +22,7 @@ const config = defineConfig({
   ],
   optimizeDeps: {
     exclude: ['kzg-wasm'],
+    include: ['vite-plugin-node-polyfills/shims/buffer', 'vite-plugin-node-polyfills/shims/global', 'vite-plugin-node-polyfills/shims/process']
```
Comment (holgerd77) [inline config/vitest.config.browser.mts:25]: At first look, these additions seem to go against our as-pure-as-possible-browser-support direction strategy. 🤔 

Is this something to further investigate (why needed, what changed, where to adjust)?

(I just created a non-minifed VM (so not: EVM) bundle as a first test and searched for "Buffer" but did not find anything suspicious)

Comment (acolytec3) [reply config/vitest.config.browser.mts:25]: I would agree that this was my surface thought as well.  We have these various shims by default as a part of the `vite-plugin-node-polyfills` dep for running our tests and vitest was having trouble reloading them or something (saw these messages about optimizing these deps over and over again in the logs) and the suggested fix was to add them to the `optimizeDeps` list in our config.  Given the effort on getting rid of polyfills, we might be able to just get rid of that plugin though I'm not 100% sure.  Worth researching.

**Pull request merged**